packages/api/cms-api/src/generator/generate-crud-input.ts

```diff
@@ -313,6 +316,32 @@ export async function generateCrudInput(
             console.warn(`${prop.name}: unsupported type ${type}`);
             continue;
         }
+
+        const classValidatorValidators = getMetadataStorage().getTargetValidationMetadatas(metadata.class, prop.name, false, false, undefined);
+        for (const validator of classValidatorValidators) {
+            if (validator.propertyName !== prop.name) continue;
+            const constraints = getMetadataStorage().getTargetValidatorConstraints(validator.constraintCls);
+            for (const constraint of constraints) {
+                const decorator = definedDecorators.find((decorator) => {
+                    return (
+                        // ignore casing since class validator is inconsistent with casing
+                        decorator.getName().toUpperCase() === constraint.name.toUpperCase() ||
+                        // some class validator decorators have a prefix "Is" but not in the constraint name
+                        `Is${decorator.getName()}`.toUpperCase() === constraint.name.toUpperCase()
+                    );
+                });
+                if (decorator) {
+                    const importPath = findValidatorImportPath(decorator.getName(), generatorOptions, metadata);
+                    if (importPath) {
+                        imports.push({ name: decorator.getName(), importPath });
+                        decorators.includes(decorator.getText()) || decorators.unshift(decorator.getText());
```

Comment:
Nit: this would be easier to comprehend when done in a "normal" if-condition

Also, why would you add this at the beginning? 🤔

Reply:
It's added in the beginning so the generated input has the same decorator order as the defined decorators in the entity and is on top of any automatically generated decorators. Imo the generated code makes more sense this way
 
entity: 
```
    @IsEmail()
    @Length(4, 5)
    @Property({ columnType: "text" })
    email: string;
```

generated input: 

```
 export class TestEntityWithEmailInput {
                 @IsEmail()
                 @Length(4, 5)
                 @IsNotEmpty()
                 @IsString()
                 @Field()
                 email: string;
              }
```

Reply:
Also changed this to an if condition in [25d9cb9](https://github.com/vivid-planet/comet/pull/1828/commits/25d9cb9ee52779ccb54c7b13b2697cd392c8a319)